probe-rs/src/core/registers.rs

```diff
@@ -440,39 +489,52 @@ impl CoreRegisters {
     pub fn msp(&self) -> Option<&CoreRegister> {
         self.0
             .iter()
-            .find(|r| r.role == Some(RegisterRole::MainStackPointer))
+            .find(|r| r.register_has_role(RegisterRole::MainStackPointer))
             .cloned()
     }
 
     /// The process stack pointer.
     pub fn psp(&self) -> Option<&CoreRegister> {
         self.0
             .iter()
-            .find(|r| r.role == Some(RegisterRole::ProcessStackPointer))
+            .find(|r| r.register_has_role(RegisterRole::ProcessStackPointer))
             .cloned()
     }
 
     /// The processor status register.
     pub fn psr(&self) -> Option<&CoreRegister> {
         self.0
             .iter()
-            .find(|r| r.role == Some(RegisterRole::ProcessorStatus))
+            .find(|r| r.register_has_role(RegisterRole::ProcessorStatus))
             .cloned()
     }
 
-    /// Other architecture specific registers
+    /// Return an iterator over all the architecture specific registers which do not have a specific role.
     pub fn other(&self) -> impl Iterator<Item = &CoreRegister> {
         self.0
             .iter()
-            .filter(|r| matches!(r.role, Some(RegisterRole::Other(_))))
+            .filter(|r| {
+                if let Some(role_list) = r.roles {
+                    let mut is_other_register = false;
+                    for role in role_list {
+                        if matches!(role, RegisterRole::Other(_)) {
```

Comment:
This seems to contradict the documentation. If a register has both `RegisterRole::Other` and another rule, should it be returned by this function?

Reply:
@Tiwalun, I think my doc comment (everwhere I talk about `RegisterRole::Other` is misleading. I will fix it. 

To answer your question, I think it should return it. Consider this RISC-V register ...
```
CoreRegister {
    name: "x8",
    roles: &[RegisterRole::FramePointer, RegisterRole::Other("s0")],
    id: RegisterId(0x1008),
    data_type: RegisterDataType::UnsignedInteger(32),
}
```

If I ask for `other_by_name("s0")` ... it should return it, even if it has other rules. 

Does that make sense?

Reply:
I guess, but I to be honest I fail to see a use case for this function.

Reply:
Now that you say it that way, I totally agree :)

Consider it gone!